cdap-docs/admin-manual/source/security/impersonation.rst

```diff
@@ -48,6 +61,113 @@ described in the installation section for each distribution (:ref:`Cloudera Mana
 <cloudera-hdfs-permissions>`, :ref:`Ambari <ambari-hdfs-permissions>`, 
 :ref:`MapR <mapr-hdfs-permissions>`, and :ref:`packages <packages-hdfs-permissions>`).
 
+Application-level Impersonation
+-------------------------------
+To use application-level impersonation in CDAP |---| where applications, datasets, and streams have
+their own owner and the operations performed in CDAP impersonate their respective
+owners |---| CDAP needs to have access to the owner principal and their associated keytabs.
+
+.. highlight:: xml
+
+For user's keytab access, CDAP uses these conventions:
+
+- All keytabs must be present on the local filesystem of nodes on which the CDAP Master is running. 
+- These keytabs must be present under a path which can be in one of these formats
+  and the ``cdap`` system user should have read access to all of the keytabs::
+
+    /<dir-1>/<dir-2>/${name}.keytab
+    /<dir-1>/<dir-2>/${name}/${name}.keytab
+
+- The above path is provided to CDAP as a configuration parameter in the ``cdap-site.xml``
+  file, such as::
+
+    <property>
+        <name>security.keytab.path</name>
+        <value>/etc/security/keytabs/${name}.keytab</value>
+    </property>
+
+  where ``${name}`` will be replaced by CDAP by the short user name of the Kerberos
+  principal CDAP is impersonating.
+  
+  **Note:** You will need to restart CDAP for this configuration change to take effect.
+
+Owner principal of an entity is provided either when an entity is created using the CDAP
+CLI or the RESTful APIs or when an application creates them.
+
+Hive Configuration
+------------------
+In order for Hive to work with impersonation, one of the following approaches can be used:
+
+- Hive Proxy Users; or
+- Hive SQL-based Authorization
+
+**Hive Proxy Users**
+
+To configure Hive to be able to impersonate other users, set in ``hive-site.xml`` the property::
+
+  <property>
+      <name>hive.server2.enable.doAs</name>
+      <value>true</value>
+  </property>
+
+Note that the CDAP Explore service ignores this setting and needs to be able to
+impersonate users who can create and access entities in CDAP. This can by done by adding
+properties in your ``core-site.xml``. The first property allows CDAP to impersonate users
```

Comment:
The first property allows hive to impersonate users belonging to the listed groups.